src/components/Products/index.jsx

```diff
@@ -0,0 +1,43 @@
+import {
+  FavoriteBorderOutlined,
+  SearchOutlined,
+  ShoppingCartOutlined,
+} from "@material-ui/icons";
+import React, { useContext } from "react";
+import { Link } from "react-router-dom";
+import { Container, Heading, Icon, Image } from "./styles";
+import { MainContext } from "../../Context";
+
+export const Products = () => {
+  const { state, dispatch } = useContext(MainContext);
+  return (
+    <>
+      {state.categories.map((item) => {
+        return (
+          <Container key={item._id}>
+            <Image src={item.image}></Image>
+            <Link
+              to="/product-list"
+              className="links"
+              onClick={(e) =>
+                dispatch({ type: "SET_CATEGORY", payload: item.categoryName })
+              }
+            >
+              <Heading>
```

Comment:
very well broken code into the components

Reply:
thanks